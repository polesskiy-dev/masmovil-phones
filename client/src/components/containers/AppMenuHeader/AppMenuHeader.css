.app-menu {
    background-color: var(--app-menu-bg-color);
    color: var(--app-menu-text-color);
    height: 56px;

    padding-left: 56px;

    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
}

.app-menu .app-menu__menu {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;

    margin: 0;

    font-size: 20px;
    list-style: none;
}

@media only screen and (max-width: 480px) {
    .app-menu {
        flex-direction: column;
        height: 100%;
        width: 100%;

        padding-left: 0;
    }

    .app-menu .app-menu__greeting {
        width: 100%;
        font-size: 24px;
    }

    .app-menu .app-menu__menu {
        width: 100%;
        flex-direction: column;
        padding-left: 0;
    }
}