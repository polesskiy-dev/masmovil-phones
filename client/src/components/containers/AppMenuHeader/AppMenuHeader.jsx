import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import './AppMenuHeader.css';

import MenuButton from '../../presentional/MenuButton/MenuButton';
import { switchLanguage } from '../../../ducks/language.duck';
import { switchTheme } from '../../../ducks/theme.duck';
import { getLanguage } from '../../../selectors/language.selector';
import { getTheme } from '../../../selectors/theme.selector';

const AppMenuHeader = ({ switchLanguage, switchTheme, language, theme }) => (
  <div className="app-menu">
    <header>
      <h1>
        <FormattedMessage id="AppMenuHeader.greeting" />
      </h1>
    </header>
    <ul className="app-menu__menu">
      <li>
        <MenuButton
          messageId="AppMenuHeader.language"
          caption={language.toUpperCase()}
          onClick={switchLanguage}
        />
      </li>
      <li>
        <MenuButton
          messageId="AppMenuHeader.language"
          caption={theme.toUpperCase()}
          onClick={switchTheme}
        />
      </li>
    </ul>
  </div>
);

AppMenuHeader.propTypes = {
  switchLanguage: PropTypes.func.isRequired,
  theme: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
};

export default connect(
  createSelector(getLanguage, getTheme, (language, theme) => ({
    language,
    theme,
  })),
  { switchLanguage, switchTheme }
)(AppMenuHeader);
