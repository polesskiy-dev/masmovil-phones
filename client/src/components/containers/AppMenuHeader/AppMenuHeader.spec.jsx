import React from 'react';
import renderer from 'react-test-renderer';
import _ from 'lodash/fp';

import withDefaultIntl from '../../../helpers/with-default-intl';
import { AppMenuHeaderRaw } from './AppMenuHeader';

it('<AppMenuHeaderRaw/ > renders correctly with all required fields', () => {
  const tree = renderer
    .create(
      withDefaultIntl(
        <AppMenuHeaderRaw
          language="en"
          switchLanguage={_.noop}
          theme="light"
          switchTheme={_.noop}
        />
      )
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
