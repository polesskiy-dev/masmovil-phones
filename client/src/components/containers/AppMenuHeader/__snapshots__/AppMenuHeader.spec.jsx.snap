// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppMenuHeaderRaw/ > renders correctly with all required fields 1`] = `
<div
  className="app-menu"
>
  <header>
    <h1
      className="app-menu__greeting"
    >
      <span>
        Welcome!
      </span>
    </h1>
  </header>
  <ul
    className="app-menu__menu"
  >
    <li>
      <div
        className="menu-button__wrapper"
      >
        <button
          className="menu-button"
          onClick={[Function]}
          type="button"
        >
          <span>
            Language:
          </span>
           
          EN
        </button>
      </div>
    </li>
    <li>
      <div
        className="menu-button__wrapper"
      >
        <button
          className="menu-button"
          onClick={[Function]}
          type="button"
        >
          <span>
            Language:
          </span>
           
          LIGHT
        </button>
      </div>
    </li>
  </ul>
</div>
`;
